libraries/AP_Mount/AP_Mount_Backend.cpp

```diff
@@ -414,6 +427,121 @@ void AP_Mount_Backend::write_log(uint64_t timestamp_us)
+        // run this loop at 10hz
+        hal.scheduler->delay(100);
+
+        // get the current location of vehicle
+        const AP_AHRS &ahrs = AP::ahrs();
+        Location curr_loc;
+        if (!ahrs.get_location(curr_loc)) {
+            continue;
+        }
+
+        // change vehicle location to AMSL
+        curr_loc.change_alt_frame(Location::AltFrame::ABSOLUTE);
+
+        // retrieve gimbal attitude
+        Quaternion quat;
+        if (!get_attitude_quaternion(quat)) {
+            // gimbal attitude unavailable
+            continue;
+        }
+
+        // project forward from vehicle looking for terrain
+        // start testing at vehicle's location
+        Location test_loc = curr_loc;
+        Location prev_test_loc = curr_loc;
+
+        auto terrain = AP_Terrain::get_singleton();
+        float terrain_amsl_m;
+        // get terrain altitude (AMSL) at test_loc
+        if (!terrain->height_amsl(test_loc, terrain_amsl_m, true)) {
+            // failed to get terrain altitude
+            continue;
+        }
+
+        float dist_increment_m;
+        if (!terrain->get_terrain_spacing(dist_increment_m)) {
+            // failed to get the terrain spacing
+            continue;
+        }
+
+        float dist_max = 10000; // Hardcoded for testing. probably need a parameter
+        float total_dist = 0;
+        float mount_pitch_deg = degrees(quat.get_euler_pitch());
+        float mount_yaw_ef_deg = wrap_180(degrees(quat.get_euler_yaw()) + degrees(ahrs.get_yaw()));
+        bool get_terrain_alt_success = true;
+        float prev_terrain_amsl_m = terrain_amsl_m;
+        // iteratively move test_loc forward until its alt-above-sea-level is below terrain-alt-above-sea-level
+        while (total_dist < dist_max && (test_loc.alt * 0.01) > terrain_amsl_m) {
+            total_dist += dist_increment_m;
+
+            // Take backup of previous test location and terrain amsl
+            prev_test_loc = test_loc;
+            prev_terrain_amsl_m = terrain_amsl_m;
+
+            // Move test location forward
+            test_loc.offset_bearing_and_pitch(mount_yaw_ef_deg, mount_pitch_deg, dist_increment_m);
+
+            // Get terrain's alt-above-sea-level (at test_loc)
+            // Fail if terrain alt cannot be retrieved
+            if (!terrain->height_amsl(test_loc, terrain_amsl_m, true) || std::isnan(terrain_amsl_m)) {
+                // failed to get terrain alt
+                get_terrain_alt_success = false;
+                continue;
+            }
+        }
+        // if a fail occurs in the loop while getting terrain alt then continue to next interation
+        if (!get_terrain_alt_success) {
+            continue;
+        }
+        if (total_dist >= dist_max) {
+            // POI: unable to find terrain within dist_max
+            continue;
+        } else if (is_negative(terrain_amsl_m)) {
+            // POI: failed to retrieve terrain alt
+            continue;
+        } else {
```

Comment:
```suggestion
        }
```

.. and reindent and remove extra brace

Reply:
done